### ISSUE TITLE
Plans for Episode Season Detection?

### ISSUE THREAD

**Issue by wax911** [igorcmoura/anitopy#3]
I was wondering if you have any plans to add seasonal information in titles? e.g:
```
[HorribleSubs] Mob Psycho 100 S2 - 07 [1080p].mkv
```

Thank you for this neat and handy tool, really saved me from a lot of headaches

**Comment by Euklios** [comment]
I've already come across the same limitation and actually think it's a good suggestion.
So I just started to implement this one.

**Comment by wax911** [comment]
:smiley: that's great to hear!! looking forward to it, thanks again for your time!

**Comment by Euklios** [comment]
I was able to implement this behavior so far, thus season 2 and episode 07 are read from the given file path.
But I didn't change the title, because I assume that the compatibility with other code can be better maintained. So the complete result is:
```python
{
    'file_name': '[HorribleSubs] Mob Psycho 100 S2 - 07 [1080p].mkv',
    'file_extension': 'mkv',
    'video_resolution': '1080p',
    'anime_season': '2',
    'episode_number': '07',
    'anime_title': 'Mob Psycho 100 S2',
    'release_group': 'HorribleSubs'
}
```

I have not adjusted the test data yet, so the tests will not pass for the following files:
```
[Conclave-Mendoi]_Mobile_Suit_Gundam_00_S2_-_01v2_[1280x720_H.264_AAC][4863FBE8].mkv
Ookiku Furikabutte S2 - 09 (Central Anime) [BD841253].mkv
[SFW]_Queen\'s_Blade_S2
[Hatsuyuki]_Kuroko_no_Basuke_S3_-_01_(51)_[720p][10bit][619C57A0].mkv
```

All of them, because now the season number is also available in the result.

**Comment by igorcmoura** [comment]
Oh, I didn't notice it was not covering this case. Thanks for pointing out @wax911. Also thanks for the pull request @Euklios.

I copied the test cases from Anitomy and I'm not entirely sure why erengy decided not to parse seasons in this format. Maybe because there are some animes with `S2` in the name? I don't know, I need an example.

Anyway, just like it parses `Season 2` or `2nd Season` it should parse `S2` and remove it from the title. I think cases where the season has to be in the title are exceptions and should be handled by the application using this module, and not the module itself.

What do you think?

**Comment by igorcmoura** [comment]
I asked erengy why he decided to take this approach. Lets see if we're not missing something here...

**Comment by igorcmoura** [comment]
No answer so far 😕 

Lets follow the approach of parsing the `S#` format. @Euklios could you change in the Pull Request to remove the `S#` from the title?

**Comment by igorcmoura** [comment]
Also, how do you think it should be handled if the filename was something like this?
```
[HorribleSubs] Mob Psycho 100 S2-07 [1080p].mkv
```

**Comment by wax911** [comment]
By handled do you mean the technical aspect of it all, or just output? If output perhaps something like this? :thinking: 

```json
{
    "file_name": "[HorribleSubs] Mob Psycho 100 S2-07 [1080p].mkv", 
    "file_extension": "mkv", 
    "video_resolution": "1080p", 
    "episode_number": "07", 
    "anime_title": "Mob Psycho 100", 
    "release_group": "HorribleSubs", 
    "season_number": 2
}
```

**Comment by igorcmoura** [comment]
Yeah, I meant the output. Because in @Euklios' pull request the parser will handle the `07` as another season.

I will merge the PR and then fix these issues. Also, sorry for the late response.

**Comment by igorcmoura** [comment]
I released version 2.0.0 with these changes.

Unfortunately parsing the season and the episode in the same token (e.g.: `S2-07`) turned out to be more complex than I expected. I'll have to refactor the code before doing it, so I just left this case unresolved for now.
